core/web/job_specs_controller.go

```diff
@@ -49,25 +49,45 @@ func (jsc *JobSpecsController) requireImplented(js models.JobSpec) error {
 	return nil
 }
 
+// getAndCheckJobSpec(c) returns a validated job spec from c, or errors. On
+// error, a non-zero HTTP status is returned.
+func (jsc *JobSpecsController) getAndCheckJobSpec(
+	c *gin.Context) (js models.JobSpec, httpStatus int, err error) {
+	var jsr models.JobSpecRequest
+	if err := c.ShouldBindJSON(&jsr); err != nil {
+		// TODO(alx): Better parsing and more specific error messages
+		// https://www.pivotaltracker.com/story/show/171164115
+		return models.JobSpec{}, http.StatusBadRequest, err
+	}
+	js = models.NewJobFromRequest(jsr)
+	if err := jsc.requireImplented(js); err != nil {
+		return models.JobSpec{}, http.StatusNotImplemented, err
+	}
+	if err := services.ValidateJob(js, jsc.App.GetStore()); err != nil {
+		return models.JobSpec{}, http.StatusBadRequest, err
+	}
+	return js, 0, nil
+}
```

Comment:
Might be a subjective thing, but having the http Status codes and their conditions within a function makes it easier to read.

Reply:
Can you give me an example? I don't follow.

Reply:
Most of the returned HTTP status codes are contained in one function ->
https://github.com/smartcontractkit/chainlink/blob/develop/core/web/job_runs_controller.go#L53-L69

Why not just include this in Create?  I'm unsure what's being saved here.